src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -80,19 +78,21 @@ object Content {
    * @param hadoopConfiguration Hadoop configuration.
    * @param pathFilter Filter for accepting paths. The default filter is picked from spark
    *                   codebase, which filters out files like _SUCCESS.
-   * @param throwIfNotExists Throws FileNotFoundException if path is not found. Else creates a
-   *                         blank directory tree with no files.
    * @return Content object with Directory tree starting at root, and containing all leaf files
-   *         from "path" argument.
+   *         from "path" argument. If the given path does not exist, return Content object with
+   *         empty Directory object that represents the path.
```

Comment:
Nit: Although it's easy to guess, the wording here isn't clear about whether the empty directory will be created on the file system. The method name "fromDirectory" sounds like it's a read-only action, but now it can modify the file system and it could be surprising to developers.

Reply:
It's still read-only function; just creating a `Directory` object, not a real directory.